full/src/main/java/apoc/couchbase/CouchbaseConnection.java

```diff
@@ -32,182 +43,182 @@
+            return null;
+        }
     }
 
     /**
-     * Checks whether a {@link JsonDocument} with the given ID does exist.
+     * Checks whether a document with the given ID does exist.
      *
      * @param documentId the unique ID of the document
      * @return true if it exists, false otherwise.
-     * @see Bucket#exists(String)
+     * @see Collection#exists(String)
      */
     public boolean exists(String documentId) {
-        return this.bucket.exists(documentId);
+        return this.collection.exists(documentId).exists();
     }
 
     /**
-     * Inserts a {@link JsonDocument} if it does not exist already.
+     * Inserts a full document which does not exist yet.
      *
      * @param documentId the unique ID of the document
      * @param json       the JSON String representing the document to store
-     * @return the newly created {@link JsonDocument}
-     * @see Bucket#insert(Document)
+     * @return the newly created {@link MutationResult}
+     * @see Collection#insert(String, Object)
      */
-    public JsonDocument insert(String documentId, String json) {
-        return this.bucket.insert(JsonDocument.create(documentId, JsonObject.fromJson(json)));
+    public MutationResult insert(String documentId, String json) {
+        return this.collection.insert(documentId, JsonObject.fromJson(json));
     }
 
     /**
-     * Inserts or overwrites a {@link JsonDocument}.
+     * Inserts or overwrites a document.
      *
      * @param documentId the unique ID of the document
      * @param json       the JSON String representing the document to store
-     * @return the newly created or overwritten {@link JsonDocument}
-     * @see Bucket#upsert(Document)
+     * @return the newly created or overwritten {@link MutationResult}
+     * @see Collection#upsert(String, Object)
      */
-    public JsonDocument upsert(String documentId, String json) {
-        return this.bucket.upsert(JsonDocument.create(documentId, JsonObject.fromJson(json)));
+    public MutationResult upsert(String documentId, String json) {
+        return this.collection.upsert(documentId, JsonObject.fromJson(json));
     }
 
     /**
-     * Appends a {@link JsonDocument}'s content to an existing one.
+     * Appends a json content to an existing one.
      *
      * @param documentId the unique ID of the document
      * @param json       the JSON String representing the document to append
-     * @return the updated {@link Document}
-     * @see Bucket#append(Document)
+     * @return the updated {@link MutationResult}
+     * @see BinaryCollection#append(String, byte[])
      */
-    public JsonDocument append(String documentId, String json) {
-        this.bucket.append(JsonDocument.create(documentId, JsonObject.fromJson(json)));
-        return this.get(documentId);
+    public MutationResult append(String documentId, String json) {
+        return binaryCollection.append(documentId, JsonObject.fromJson(json).toBytes());
```

Comment:
@fbiville oh, sorry about that, I implemented sdk 3 version, assuming the old version was right, that is with a json. Actually, the `Bucket.append(document)` of the sdk2 says: "Note that JsonDocuments in all forms are not supported", so it wasn't right before. 

Now I have changed append and prepend from `JsonObject.fromJson(json).toBytes()` to `json.getBytes()` and added test. 

Does it make sense, at this point, to also change the name of the procedure parameter from `"json"` to `"content"` (consistent with `BinaryCollection.append(..)`) and to directly accept `byte[]` instead of `String`?

Reply:
Good idea, `content` is a better name indeed.